### PR TITLE
Move ObjectMeta into pkg/apis/meta/v1

### PR DESCRIPTION
**Pull request by smarterclayton**
Where it belongs

### CODE REVIEW

File: pkg/api/resource_helpers.go [hunk 1]
```diff
@@ -231,7 +231,7 @@ func PodRequestsAndLimits(pod *Pod) (reqs map[ResourceName]resource.Quantity, li
 
 // ValidNamespace returns false if the namespace on the context differs from the resource.  If the resource has no namespace, it is set to the value in the context.
 // TODO(sttts): move into pkg/genericapiserver/api
-func ValidNamespace(ctx genericapirequest.Context, resource *ObjectMeta) bool {
```
Comment (sttts) [inline pkg/api/resource_helpers.go]: moving this to pkg/api/rest in https://github.com/kubernetes/kubernetes/pull/39729, whichever merges first....

File: cmd/libs/go2idl/conversion-gen/generators/conversion.go [hunk 2]
```diff
@@ -657,7 +657,7 @@ func (g *genConversion) doSlice(inType, outType *types.Type, sw *generator.Snipp
 			} else if g.convertibleOnlyWithinPackage(inType.Elem, outType.Elem) {
 				sw.Do("if err := "+nameTmpl+"(&(*in)[i], &(*out)[i], s); err != nil {\n", argsFromType(inType.Elem, outType.Elem))
 			} else {
-				// TODO: This triggers on v1.ObjectMeta <-> api.ObjectMeta and
+				// TODO: This triggers on metav1.ObjectMeta <-> metav1.ObjectMeta and
```
Comment (sttts) [inline cmd/libs/go2idl/conversion-gen/generators/conversion.go]: obsolete comment?

Comment (sttts) [inline cmd/libs/go2idl/conversion-gen/generators/conversion.go]: outdated comment

**Comment by sttts** [comment]
Two small things, otherwise lgtm

File: pkg/api/resource_helpers.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/api/resource_helpers.go]: Ok - I'll probably wait for that to land.

File: cmd/kube-aggregator/pkg/apis/apiregistration/v1alpha1/types.generated.go [hunk 3]
```diff
@@ -1244,7 +1242,13 @@ func (x *APIService) CodecEncodeSelf(e *codec1978.Encoder) {
 				z.EncSendContainerState(codecSelfer_containerArrayElem1234)
 				if yyq94[2] {
 					yy102 := &x.ObjectMeta
-					yy102.CodecEncodeSelf(e)
+					yym103 := z.EncBinary()
+					_ = yym103
+					if false {
+					} else if z.HasExtensions() && z.EncExt(yy102) {
+					} else {
+						z.EncFallback(yy102)
```
Comment (sttts) [inline cmd/kube-aggregator/pkg/apis/apiregistration/v1alpha1/types.generated.go]: We have to create ugorji codecs for meta/v1 now that we have real structs there.

**Comment by sttts** [comment]
Note the ugorji comment. Otherwise, the slow fallback codec is used by ugorji.

File: pkg/api/resource_helpers.go (hunk 1 above)
Comment (sttts) [inline pkg/api/resource_helpers.go]: landed

**Comment by deads2k** [comment]
this has turned into rebase insanity.  Bumping priority

**Comment by deads2k** [comment]
@k8s-bot kops aws e2e test this

**Comment by deads2k** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by smarterclayton** [comment]
Hrm, bot seems confused.

@k8s-bot cvm gce e2e test this

**Comment by deads2k** [comment]
> Hrm, bot seems confused.
> 
> @k8s-bot cvm gce e2e test this

Wasn't that test green a second ago?  If so, you just needed to wait for the next refresh cycle.

**Comment by smarterclayton** [comment]
It was, but another change merged.

**Comment by deads2k** [comment]
Every pull that merges is bumping this one, update to "head of the queue" priority.  It'll probably still never get merged :)

**Comment by smarterclayton** [comment]
Unfortunately we've found the edge case - time to get this back in the queue is T(fix) + T(build).  But every other PR is just T(build).

**Comment by deads2k** [comment]
@smarterclayton you ballsey enough to push the button on green and fix the crater?

**Comment by smarterclayton** [comment]
Naw, I'll just keep playing chicken with the queue.

**Comment by smarterclayton** [comment]
So close.

**Comment by deads2k** [comment]
That button looks really green. :)

**Comment by sttts** [comment]
🚀

**Pull request merged**